src/components/Home/index.js

```diff
@@ -12,18 +12,24 @@ class HomePage extends Component {
     this.state = {
       users: null,
     };
+    this.unsubscribe = null;
   }
 
   componentDidMount() {
-    this.props.firebase.users().on('value', snapshot => {
-      this.setState({
-        users: snapshot.val(),
+    this.unsubscribe = this.props.firebase
+      .users()
+      .onSnapshot(snapshot => {
+        let users = {};
+        snapshot.forEach(doc => (users[doc.id] = doc.data()));
```

Comment:
Side Comment: Interesting that they made this the other way around for the Firestore API where the result is an array and not an object anymore.

Reply:
Suggestion: If you want to allocate something into an object, `reduce` is also a good way to do it:

```
const users = snapshot.reduce((result, value, key) => {
  result[doc.id] = doc.data();
  return result;
}, {});
```

Every time forEach is used, there may be a way to use map, reduce or filter instead. The great thing about the 3 functions is that they always return a new object/array whereas `forEach` could tempt one to modify the source array.

Reply:
Yes, I absolutely agree with you, but unfortunately QuerySnapshot doesn't have `map`, `reduce`, only `forEach` for iterating objects.